In [279]:
import pandas as pd

In [280]:
ds=pd.read_csv("quikr_car.csv")

In [281]:
ds

name    company  year          Price  \
0      Hyundai Santro Xing XO eRLX Euro III    Hyundai  2007         80,000   
1                   Mahindra Jeep CL550 MDI   Mahindra  2006       4,25,000   
2                Maruti Suzuki Alto 800 Vxi     Maruti  2018  Ask For Price   
3    Hyundai Grand i10 Magna 1.2 Kappa VTVT    Hyundai  2014       3,25,000   
4          Ford EcoSport Titanium 1.5L TDCi       Ford  2014       5,75,000   
..                                      ...        ...   ...            ...   
887                                      Ta       Tara  zest       3,10,000   
888                     Tata Zest XM Diesel       Tata  2018       2,60,000   
889                      Mahindra Quanto C8   Mahindra  2013       3,90,000   
890                Honda Amaze 1.2 E i VTEC      Honda  2014       1,80,000   
891               Chevrolet Sail 1.2 LT ABS  Chevrolet  2014       1,60,000   

     kms_driven fuel_type  
0    45,000 kms    Petrol  
1        40 kms    Diesel  
2    22,000 kms    Petrol  
3    28,000 kms    Petrol  
4    36,000 kms    Diesel  
..          ...       ...  
887         NaN       NaN  
888  27,000 kms    Diesel  
889  40,000 kms    Diesel  
890      Petrol       NaN  
891      Petrol       NaN  

[892 rows x 6 columns]

In [282]:
ds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 892 entries, 0 to 891
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   name        892 non-null    object
 1   company     892 non-null    object
 2   year        892 non-null    object
 3   Price       892 non-null    object
 4   kms_driven  840 non-null    object
 5   fuel_type   837 non-null    object
dtypes: object(6)
memory usage: 41.9+ KB


# let know problems
# 1) 1st we work on numerical column
# 2) check unique value using ds['year'].unique()
# 3) Change or delete null value or rows, remove comma and kms from kms_driven column
# 4) Change Data Type 
# 5) split Name column 
# 6) duplicate remove


# Work on year Column 

In [283]:
ds['year'].unique()

array(['2007', '2006', '2018', '2014', '2015', '2012', '2013', '2016',
       '2010', '2017', '2008', '2011', '2019', '2009', '2005', '2000',
       '...', '150k', 'TOUR', '2003', 'r 15', '2004', 'Zest', '/-Rs',
       'sale', '1995', 'ara)', '2002', 'SELL', '2001', 'tion', 'odel',
       '2 bs', 'arry', 'Eon', 'o...', 'ture', 'emi', 'car', 'able', 'no.',
       'd...', 'SALE', 'digo', 'sell', 'd Ex', 'n...', 'e...', 'D...',
       ', Ac', 'go .', 'k...', 'o c4', 'zire', 'cent', 'Sumo', 'cab',
       't xe', 'EV2', 'r...', 'zest'], dtype=object)

In [284]:
ds=ds[ds['year'].str.isnumeric()] # only take int no from column
ds['year'].unique() # check str value remove or not 

array(['2007', '2006', '2018', '2014', '2015', '2012', '2013', '2016',
       '2010', '2017', '2008', '2011', '2019', '2009', '2005', '2000',
       '2003', '2004', '1995', '2002', '2001'], dtype=object)

In [285]:
ds['year']=ds['year'].astype(int) # Change data type of year column 

C:\Users\admin\AppData\Local\Temp\ipykernel_344\3831998764.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ds['year']=ds['year'].astype(int) # Change data type of year column


# work on Price column 

In [286]:
ds=ds[ds['Price']!='Ask For Price'] # take value price not equal to 'ask for price'

In [287]:
ds['Price']=ds['Price'].str.replace(',','') # Removing Comma from column

# work on kms_driven column

In [288]:
ds

name    company  year   Price  \
0      Hyundai Santro Xing XO eRLX Euro III    Hyundai  2007   80000   
1                   Mahindra Jeep CL550 MDI   Mahindra  2006  425000   
3    Hyundai Grand i10 Magna 1.2 Kappa VTVT    Hyundai  2014  325000   
4          Ford EcoSport Titanium 1.5L TDCi       Ford  2014  575000   
6                                 Ford Figo       Ford  2012  175000   
..                                      ...        ...   ...     ...   
886                    Toyota Corolla Altis     Toyota  2009  300000   
888                     Tata Zest XM Diesel       Tata  2018  260000   
889                      Mahindra Quanto C8   Mahindra  2013  390000   
890                Honda Amaze 1.2 E i VTEC      Honda  2014  180000   
891               Chevrolet Sail 1.2 LT ABS  Chevrolet  2014  160000   

       kms_driven fuel_type  
0      45,000 kms    Petrol  
1          40 kms    Diesel  
3      28,000 kms    Petrol  
4      36,000 kms    Diesel  
6      41,000 kms    Diesel  
..            ...       ...  
886  1,32,000 kms    Petrol  
888    27,000 kms    Diesel  
889    40,000 kms    Diesel  
890        Petrol       NaN  
891        Petrol       NaN  

[819 rows x 6 columns]

In [289]:
ds['kms_driven']=ds['kms_driven'].str.split().str.get(0) # remove kms object from column 

In [290]:
ds['kms_driven']=ds['kms_driven'].str.replace(',','') # remove or replace comma from column

In [291]:
ds

name    company  year   Price  \
0      Hyundai Santro Xing XO eRLX Euro III    Hyundai  2007   80000   
1                   Mahindra Jeep CL550 MDI   Mahindra  2006  425000   
3    Hyundai Grand i10 Magna 1.2 Kappa VTVT    Hyundai  2014  325000   
4          Ford EcoSport Titanium 1.5L TDCi       Ford  2014  575000   
6                                 Ford Figo       Ford  2012  175000   
..                                      ...        ...   ...     ...   
886                    Toyota Corolla Altis     Toyota  2009  300000   
888                     Tata Zest XM Diesel       Tata  2018  260000   
889                      Mahindra Quanto C8   Mahindra  2013  390000   
890                Honda Amaze 1.2 E i VTEC      Honda  2014  180000   
891               Chevrolet Sail 1.2 LT ABS  Chevrolet  2014  160000   

    kms_driven fuel_type  
0        45000    Petrol  
1           40    Diesel  
3        28000    Petrol  
4        36000    Diesel  
6        41000    Diesel  
..         ...       ...  
886     132000    Petrol  
888      27000    Diesel  
889      40000    Diesel  
890     Petrol       NaN  
891     Petrol       NaN  

[819 rows x 6 columns]

#  Work on name column

In [292]:
ds['name'].unique()

array(['Hyundai Santro Xing XO eRLX Euro III', 'Mahindra Jeep CL550 MDI',
       'Hyundai Grand i10 Magna 1.2 Kappa VTVT',
       'Ford EcoSport Titanium 1.5L TDCi', 'Ford Figo', 'Hyundai Eon',
       'Ford EcoSport Ambiente 1.5L TDCi',
       'Maruti Suzuki Alto K10 VXi AMT', 'Skoda Fabia Classic 1.2 MPI',
       'Maruti Suzuki Stingray VXi', 'Hyundai Elite i20 Magna 1.2',
       'Mahindra Scorpio SLE BS IV', 'Audi A8', 'Audi Q7',
       'Mahindra Scorpio S10', 'Maruti Suzuki Alto 800',
       'Maruti Suzuki Alto 800 Vxi', 'Hyundai i20 Sportz 1.2',
       'Maruti Suzuki Alto 800 Lx', 'Maruti Suzuki Vitara Brezza ZDi',
       'Maruti Suzuki Alto LX', 'Mahindra Bolero DI',
       'Maruti Suzuki Swift Dzire ZDi', 'Mahindra Scorpio S10 4WD',
       'Maruti Suzuki Swift Vdi BSIII',
       'Maruti Suzuki Wagon R VXi BS III',
       'Maruti Suzuki Wagon R VXi Minor',
       'Toyota Innova 2.0 G 8 STR BS IV', 'Renault Lodgy 85 PS RXL',
       'Skoda Yeti Ambition 2.0 TDI CR 4x2',
       'Maru

In [293]:
ds['name']=ds['name'].str.split().str.get(1) # remove compony name from car names 
ds

name    company  year   Price kms_driven fuel_type
0      Santro    Hyundai  2007   80000      45000    Petrol
1        Jeep   Mahindra  2006  425000         40    Diesel
3       Grand    Hyundai  2014  325000      28000    Petrol
4    EcoSport       Ford  2014  575000      36000    Diesel
6        Figo       Ford  2012  175000      41000    Diesel
..        ...        ...   ...     ...        ...       ...
886   Corolla     Toyota  2009  300000     132000    Petrol
888      Zest       Tata  2018  260000      27000    Diesel
889    Quanto   Mahindra  2013  390000      40000    Diesel
890     Amaze      Honda  2014  180000     Petrol       NaN
891      Sail  Chevrolet  2014  160000     Petrol       NaN

[819 rows x 6 columns]

In [295]:
ds.dropna()

name   company  year   Price kms_driven fuel_type
0      Santro   Hyundai  2007   80000      45000    Petrol
1        Jeep  Mahindra  2006  425000         40    Diesel
3       Grand   Hyundai  2014  325000      28000    Petrol
4    EcoSport      Ford  2014  575000      36000    Diesel
6        Figo      Ford  2012  175000      41000    Diesel
..        ...       ...   ...     ...        ...       ...
883    Suzuki    Maruti  2011  270000      50000    Petrol
885    Indica      Tata  2009  110000      30000    Diesel
886   Corolla    Toyota  2009  300000     132000    Petrol
888      Zest      Tata  2018  260000      27000    Diesel
889    Quanto  Mahindra  2013  390000      40000    Diesel

[816 rows x 6 columns]

In [299]:
ds.drop_duplicates(inplace=True)

In [306]:
ds

name    company  year   Price kms_driven fuel_type
0      Santro    Hyundai  2007   80000      45000    Petrol
1        Jeep   Mahindra  2006  425000         40    Diesel
3       Grand    Hyundai  2014  325000      28000    Petrol
4    EcoSport       Ford  2014  575000      36000    Diesel
6        Figo       Ford  2012  175000      41000    Diesel
..        ...        ...   ...     ...        ...       ...
886   Corolla     Toyota  2009  300000     132000    Petrol
888      Zest       Tata  2018  260000      27000    Diesel
889    Quanto   Mahindra  2013  390000      40000    Diesel
890     Amaze      Honda  2014  180000     Petrol       NaN
891      Sail  Chevrolet  2014  160000     Petrol       NaN

[721 rows x 6 columns]

# Now we successfully clean Data set of Quikr car data
# We found 721 rows * 6 column 